omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import static org.openmrs.ConditionVerificationStatus.CONFIRMED;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_2;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link DiagnosisController2_2}.
+ */
+public class DiagnosisController2_2Test extends MainResourceControllerTest {
+
+	private DiagnosisService diagnosisService;
+
+	private Encounter encounter;
+
+	@Before
+	public void before() throws Exception {
+		executeDataSet(RestTestConstants2_2.DIAGNOSIS_TEST_DATA_XML);
+		this.diagnosisService = Context.getDiagnosisService();
+		this.encounter = new Encounter();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return RestConstants.VERSION_1 + "/patientdiagnoses";
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.DIAGNOSIS_UUID;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return diagnosisService.getPrimaryDiagnoses(this.encounter).size();
+	}
+
+	@Test
+	public void shouldCreateADiagnosis() throws Exception {
+		long originalCount = getAllCount();
+
+		SimpleObject diagnosisSource = new SimpleObject();
+		diagnosisSource.add("diagnosisId", 1);
+		diagnosisSource.add("encounter", new Encounter());
```

Comment:
The above line is the cause of the error that you are getting. Instead of an Encounter object, it should be a uuid string for the encounter. Can you talk to @Arthur236 to help you in regards to this?